imap_processing/swe/l2/swe_l2.py

```diff
@@ -113,4 +114,58 @@ def calculate_phase_space_density(l1b_dataset: xr.Dataset) -> npt.NDArray:
         * particle_energy_data[:, :, :, np.newaxis] ** 2
     )
 
-    return density
+    # Return density as xr.dataset with phase space density and
+    # energy in eV value that flux calculation can use.
+    phase_space_density_dataset = xr.Dataset(
+        {
+            "phase_space_density": (["epoch", "energy", "angle", "cem"], density.data),
```

Comment:
Oh, interesting. I hadn't seen this mapping for passing in tuples of dimension names and a numpy array.